numba/cuda/tests/cudapy/test_operator.py

```diff
@@ -105,6 +137,118 @@ def test_floordiv(self):
+            with self.subTest(op=op, ty=ty):
+                kernel = cuda.jit(fn)
+
+                arg1 = np.random.random(1).astype(np.float16)
+                arg2 = (np.random.random(1) * 100).astype(ty)
+                arg2_ty = np.result_type(np.float16, ty)
+
+                expected = np.zeros(1, dtype=arg2_ty)
+                got = np.zeros(1, dtype=arg2_ty)
+
+                kernel[1, 1](got, arg1[0], arg2[0])
+                expected = op(arg1, arg2)
+                np.testing.assert_allclose(got[0], expected)
+
+    @skip_on_cudasim('Compilation unsupported in the simulator')
+    def test_fp16_inplace_binary_ptx(self):
+        functions = (simple_fp16_iadd, simple_fp16_isub, simple_fp16_imul)
+        ops = (operator.iadd, operator.isub, operator.imul)
+
+        opstring = ('add.f16', 'sub.f16', 'mul.f16')
+        args = (f2[:], f2)
+
+        for fn, op, s in zip(functions, ops, opstring):
+            with self.subTest(op=op):
+                ptx, _ = compile_ptx(fn, args, cc=(5, 3))
+                self.assertIn(s, ptx)
+
+    @skip_unless_cc_53
+    def test_fp16_inplace_binary(self):
+        functions = (simple_fp16_iadd, simple_fp16_isub, simple_fp16_imul)
+        ops = (operator.iadd, operator.isub, operator.imul)
+
+        for fn, op in zip(functions, ops):
+            with self.subTest(op=op):
+                kernel = cuda.jit("void(f2[:], f2)")(fn)
+
+                expected = np.zeros(1, dtype=np.float16)
+                got_in_out = np.random.random(1).astype(np.float16)
+                arg1 = np.random.random(1).astype(np.float16)
+
+                kernel[1, 1](got_in_out, arg1[0])
+                expected = op(got_in_out, arg1)
+                np.testing.assert_allclose(got_in_out[0], expected)
+
+    @skip_unless_cc_53
+    def test_fp16_unary(self):
+        functions = (simple_fp16neg, simple_fp16abs)
+        ops = (operator.neg, operator.abs)
+
+        for fn, op in zip(functions, ops):
+            with self.subTest(op=op):
+                kernel = cuda.jit("void(f2[:], f2)")(fn)
+
+                expected = np.zeros(1, dtype=np.float16)
+                got = np.zeros(1, dtype=np.float16)
+                arg1 = np.random.random(1).astype(np.float16)
+
+                kernel[1, 1](got, arg1[0])
+                expected = op(arg1)
+                np.testing.assert_allclose(got[0], expected)
+
+    @skip_on_cudasim('Compilation unsupported in the simulator')
+    def test_fp16_neg_ptx(self):
+        args = (f2[:], f2)
+        ptx, _ = compile_ptx(simple_fp16neg, args, cc=(5, 3))
+        self.assertIn('neg.f16', ptx)
+
+    @skip_on_cudasim('Compilation unsupported in the simulator')
+    def test_fp16_abs_ptx(self):
+        args = (f2[:], f2)
+        ptx, _ = compile_ptx(simple_fp16abs, args, cc=(5, 3))
+        if cuda.runtime.get_version() < (10, 2):
+            self.assertRegex(ptx, r'and\.b16.*0x7FFF;')
+        else:
+            self.assertIn('abs.f16', ptx)
```

Comment:
We only support CUDA 10.2 onwards now, so the check for prior versions is not needed:
```suggestion
        self.assertIn('abs.f16', ptx)
```